concert_service_manager/src/concert_service_manager/service_cache_manager.py

```diff
@@ -0,0 +1,468 @@
+      This is not validated against the actual service here.
+
+      :param yaml_file: filename of the solution configuration file
+      :type yaml_file: str
+
+      :returns: the solution configuration data for services in this concert
+      :rtype: [ServiceData]
+
+      :raises: :exc:`concert_service_manager.InvalidSolutionConfigurationException` if the yaml provides invalid configuration
+    """
+    service_configurations = []
+
+    # read
+    override_keys = ['name', 'description', 'icon', 'priority', 'interactions', 'parameters']
+    with open(yaml_file) as f:
+        service_list = yaml.load(f)
+        for s in service_list:
+            service_data = {}
+            service_data['resource_name'] = s['resource_name']
+            loaded_overrides = s['overrides'] if 'overrides' in s else None
+            overrides = {}
+            for key in override_keys:
+                overrides[key] = loaded_overrides[key] if loaded_overrides and key in loaded_overrides.keys() else None
+            # warnings
+            if loaded_overrides:
+                invalid_keys = [key for key in loaded_overrides.keys() if key not in override_keys]
+                for key in invalid_keys:
+                    rospy.logwarn("Service Manager : invalid key in the service soln configuration yaml [%s]" % key)
+            service_data['overrides'] = copy.deepcopy(overrides)
+            service_configurations.append(service_data)
+
+    # validate
+    identifiers = []
+    for service_data in service_configurations:
+        if service_data['overrides']['name'] is not None and 'name' in service_data['overrides'].keys():
+            identifier = service_data['overrides']['name']
+        else:
+            identifier = service_data['resource_name']
+
+        if identifier in identifiers:
+            raise InvalidSolutionConfigurationException("service configuration found with duplicate names [%s]" % identifier)
+        else:
+            identifiers.append(identifier)
+    return service_configurations
+
+
+class ServiceCacheManager(object):
+
+    __slots__ = [
+        '_concert_name',            # concert name to classify cache directory
+        '_resource_name',           # service resource name. It is composed 'package name/service resource name'. ex. )chatter_concert/chatter.service
+        '_cache_service_list',     # file and directory list cached the service profile. It is used to check file modification
+        '_modification_callback',  # callback function about file modification. It is called when _cache_service_list's item is changed.
+        'service_profiles',         # dictionary data of service profile to use in service manager
+        'msg',                        # ros message regarding service profile to use in servvice manager
+    ]
+
+    def __init__(self, concert_name, resource_name, modification_callback=None):
+        self._concert_name = concert_name.strip().lower().replace(' ', '_')
```

Comment:
Shouldn't this use https://github.com/robotics-in-concert/rocon_tools/pull/73/files#diff-f384ea7500354bbc728b8b10d20f119bR38?